### PR TITLE
move content after "[AZURE.INCLUDE" section to new line

### PR DESCRIPTION
**Pull request by jsntcy**
For example,
"[AZURE.INCLUDE [active-directory-devquickstarts-switcher](../../includes/active-directory-devquickstarts-switcher.md)] test" should be converted to 
"[AZURE.INCLUDE [active-directory-devquickstarts-switcher](../../includes/active-directory-devquickstarts-switcher.md)]
test"

### CODE REVIEW

File: tools/AzureMarkdownRewriterTool/Program.cs [hunk 1]
```diff
@@ -20,11 +20,13 @@ namespace Microsoft.DocAsCode.Tools.AzureMarkdownRewriterTool
 
     internal sealed class Program
     {
-        private static readonly Regex _azureHtmlIncludeRegex = new Regex(@"^(\<br\s*\/\>)(\s*\r?\n\[AZURE\.INCLUDE)", RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.Multiline);
+        private static readonly Regex _azureHtmlIncludeWithPrefixRegex = new Regex(@"^(\<br\s*\/\>)(\s*\r?\n\[AZURE\.INCLUDE)", RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.Multiline);
 
-        public static readonly Regex _azureHtmlDefinitionRegex = new Regex(@"^( +)(\[([^\]]+)\]: *<?([^\s>]+)>?(?: +[""(]([^\n]+)["")])? *(?:\r?\n+|$))", RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.Multiline);
+        private static readonly Regex _azureHtmlIncludeWithPostfixRegex = new Regex(@"^(\[AZURE\.INCLUDE\s*\[((?:\[[^\]]*\]|[^\[\]]|\](?=[^\[]*\]))*)\]\(\s*<?([^)]*?)>?(?:\s+(['""])([\s\S]*?)\3)?\s*\)\])[\t\f ]*(\S.*)$", RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.Multiline);
```
Comment (Huachao) [inline tools/AzureMarkdownRewriterTool/Program.cs]: >(\S.*) [](start = 211, length = 6)

do we also need to move whitespaces and tabs also to the next line?

File: tools/AzureMarkdownRewriterTool/Program.cs [hunk 2]
```diff
@@ -20,11 +20,13 @@ namespace Microsoft.DocAsCode.Tools.AzureMarkdownRewriterTool
...
 
-        public static IReadOnlyList<string> SystemMarkdownFileName = new List<string> { "TOC.md" };
+        private static readonly Regex _azureHtmlDefinitionWithLeadingWhitespacesRegex = new Regex(@"^( +)(\[([^\]]+)\]: *<?([^\s>]+)>?(?: +[""(]([^\n]+)["")])? *(?:\r?\n+|$))", RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.Multiline);
+
+        private static IReadOnlyList<string> SystemMarkdownFileName = new List<string> { "TOC.md" };
```
Comment (Huachao) [inline tools/AzureMarkdownRewriterTool/Program.cs]: I have to say, the naming rule is a little strange, for private static w/o readonly

File: tools/AzureMarkdownRewriterTool/Program.cs [hunk 3]
```diff
@@ -386,8 +397,9 @@ private int Rewrite()
                             if (_isMigration)
                             {
                                 // Fixs Azure articles first in order to let docfx parse them correctly.
-                                source = FixAzureIncludeSyntax(source);
-                                source = FixAzureDefinitionSyntax(source);
+                                source = FixAzureIncludeWithPrefixSyntax(source);
+                                source = FixAzureIncludeWithPostfixSyntax(source);
+                                source = FixAzureDefinitionWithLeadingWhitespacesSyntax(source);
```
Comment (Huachao) [inline tools/AzureMarkdownRewriterTool/Program.cs]: minor: can we change this to a chaining call, like source.FixPrefix().FixPostFix().TrimWhiteSpaces(). I think it's cleaner

**Comment by Huachao** [comment]
:shipit:

File: tools/AzureMarkdownRewriterTool/Program.cs [hunk 4]
```diff
@@ -347,15 +349,24 @@ private bool CheckParameters()
...
+        private string FixAzureIncludeWithPrefixSyntax(string source)
+        {
+            return _azureHtmlIncludeWithPrefixRegex.Replace(source, "$2");
+        }
+
+        // Moves the content after "[AZURE.INCLUDE" section to the next line.
+        // For example, "[AZURE.INCLUDE [active-directory-devquickstarts-switcher](../../includes/active-directory-devquickstarts-switcher.md)] test"
+        // should be converted to 
+        // "[AZURE.INCLUDE [active-directory-devquickstarts-switcher](../../includes/active-directory-devquickstarts-switcher.md)]\r\ntest"
+        private string FixAzureIncludeWithPostfixSyntax(string source)
```
Comment (fenxu) [inline tools/AzureMarkdownRewriterTool/Program.cs]: This change will make inline include rule to block include rule. If the include thing is a token, may bring some side effect.

**Comment by fenxu** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**